secrets/localsecrets/localsecrets.go

```diff
@@ -106,16 +110,39 @@ func Base64Key(base64str string) ([32]byte, error) {
 	if err != nil {
 		return sk32, err
 	}
+	keySize := len([]byte(key))
+	if keySize != 32 {
+		return sk32, fmt.Errorf("Base64Key: secret material is %v bytes, want 32 bytes", keySize)
+	}
 	copy(sk32[:], key)
 	return sk32, nil
 }
 
 // ByteKey takes a secret key as a string and converts it
-// to a [32]byte, cropping it if necessary.
-func ByteKey(sk string) [32]byte {
+// to a [32]byte, erroring if the string is not 32 bytes of data.
+func ByteKey(sk string) ([32]byte, error) {
```

Comment:
IMHO we should drop this function (and the `stringkey` scheme) entirely. Let's make people use the `[]byte` form (or base64 of `[]byte`).

Reply:
Getting rid of the `stringkey` scheme entirely seems like it might warrant its own issue/PR, WDYT?

Reply:
If you prefer, but it seems like part and parcel of this change to me (i.e., you'd need to update/remove some of the same `stringkey` examples you're modifying in this PR).

Reply:
Opened https://github.com/google/go-cloud/issues/1980 for removing the `ByteKey` function and `stringkey` scheme.